vignettes/Surveys_Walkthrough.Rmd

```diff
@@ -510,78 +715,154 @@ population_data_age <- bind_rows(population_data_age_district_a,
+  "B_15", 100, 
+  "B_16", 100, 
+  "B_17", 100)
 ```
 
 
-Not that we have a complete and verified list of used date variables, we can run the `across()` function within `mutate()` to convert all of the dates to class "Date". Recall that date columns must be properly recognized by R as dates in order to be used effectively.  
+## Browse data
+
+This chunk includes R commands to help you familiarise yourself with your dataset.  
+
+## Data cleaning
+
+
+### Standardise dates
+
+The goal with this code chunk is to convert all date columns to Date class so that R can handle them correctly. Here, again, we need to follow the instructions for "non-Kobo" because our data dictionary is not standard for MSF. 
+
+The default code uses the data dictionary to create a `DATEVARS` object, but we will comment that section to deactivate it, and instead use the "non-Kobo" method which searches for all columns that contain the word "date" or "Date", and converts them to class Date.  
 
 To convert the dates to class "Date" we use the function `parse_date()` from the {parsedate} package. This function does not assume a standard format to the dates, and evaluates every data frame cell individually, using intelligent guessing to decide whether the date is written as Month-Day-Year, Day-Month-Year, or Year-Month-Day, etc.
 
 The result of `parse_date()` is actually class "POSIXct" (a specific type of date class), so the result is piped to `as.Date()` to convert it to a class that is more easily accepted by subsequent functions.  
 
+
 ```{r}
-## change to dates 
-## use the parse_date() function to make a first pass at date variables.
-## parse_date() produces a complicated format - so we use as.Date() to simplify
+
+## Kobo standard data --------------------------------------------------------
+## If you got your data from Kobo, use this portion of the code.
+## If not, comment this section out and use the below.
+
+## make sure all date variables are formatted as dates 
+## get the names of variables which are dates
+# DATEVARS <- study_data_dict %>% 
+#   filter(type == "date") %>% 
+#   filter(name %in% names(study_data_cleaned)) %>% 
+#   ## filter to match the column names of your data
+#   pull(name) # select date vars
+# 
+# ## find if there are date variables which are completely empty
+# EMPTY_DATEVARS <- purrr::map(DATEVARS, ~all(
+#   is.na(study_data_cleaned[[.x]])
+#   )) %>% 
+#   unlist() %>% 
+#   which()
+# 
+# ## remove exclude the names of variables which are completely emptys
+# DATEVARS <- DATEVARS[-EMPTY_DATEVARS]
+# 
+# ## change to dates 
+# ## use the parse_date() function to make a first pass at date variables.
+# ## parse_date() produces a complicated format - so we use as.Date() to simplify
+# study_data_cleaned <- study_data_cleaned %>%
+#   mutate(
+#     across(.cols = all_of(DATEVARS),
+#            .fns = ~parsedate::parse_date(.x) %>% as.Date()))
+
+
+## Non-Kobo data -------------------------------------------------------------
```

Comment:
datevars definition missing

Reply:
dont think datevars is relevant, because we did the manual str_detect() method to find date columns...? Seems to run fine.

Reply:
sorry yeah was reading the red bit